prov/sockets/src/sock_ep.c

```diff
@@ -613,6 +613,30 @@ struct fi_ops_ep sock_ctx_ep_ops = {
 	.tx_size_left = sock_tx_size_left,
 };
 
+static void sock_ep_clear_eq_list(struct dlistfd_head *list,
+				  struct fid_ep *ep_fid)
+{
+	struct sock_eq_entry *sock_eq_entry;
+	struct fi_eq_entry *eq_entry;
+	struct dlist_entry *entry;
+
+	if (!dlistfd_empty(list)) {
+		dlist_foreach(&list->list, entry) {
+			sock_eq_entry = container_of(entry, struct sock_eq_entry,
+						entry);
+			/* fi_eq_entry, fi_eq_cm_entry and fi_eq_err_entry all
+			 * have fid_t as first member */
+			eq_entry = (struct fi_eq_entry *)sock_eq_entry->event;
+
+			if (eq_entry->fid != (fid_t)ep_fid)
+				continue;
+
+			dlistfd_remove(&sock_eq_entry->entry, list);
+			free(sock_eq_entry);
```

Comment:
Of course!  :) Thanks for taking a look at this. dlist_pop_front also removes the entry. We have to remove it only if the fids match, so I've used the dlist_remove_first_match function.